### PR TITLE
Talent  be

### PR DESCRIPTION
**Pull request by Yulllia**
-fixing google authorization choosing role
- create filter with QueryBuilder for Talent UI and fixing saving setting profile

### CODE REVIEW

File: src/entities/profile/setting-profile.entity.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { Column, Entity, ManyToOne, PrimaryGeneratedColumn } from 'typeorm';
+import { ProfileEntity } from './profile.entity';
+
+@Entity('setting')
```
Comment (lyudad) [inline src/entities/profile/setting-profile.entity.ts]: I dont think we need it, we have user table

Comment (Yulllia) [reply src/entities/profile/setting-profile.entity.ts]: It's not working before not saving data to database and I'm created new entity it's like for freelancer

File: src/entities/profile/setting-profile.entity.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+export class SettingEntity {
```
Comment (lyudad) [inline src/entities/profile/setting-profile.entity.ts]: I have wrote before, we have user table we don't need this table

Comment (Yulllia) [reply src/entities/profile/setting-profile.entity.ts]: fixed

File: src/entities/profile/profile.entity.ts [hunk 3]
```diff
@@ -39,4 +50,13 @@ export class ProfileEntity {
...
   experience: ExperienceEntity[];
+
+  @OneToMany(() => User, (user) => user.userId)
+  @JoinColumn()
+  user: User[];
+
+  @Column({ type: 'integer' })
+  @JoinColumn()
+  @ManyToOne(() => User, (user) => user.id, { cascade: true })
+  userId: number;
```
Comment (lyudad) [inline src/entities/profile/profile.entity.ts:57]: why we need both user and userId, according to nest documentation userId is not needed. please be more attentive to details

File: src/entities/user.entity.ts [hunk 4]
```diff
@@ -11,12 +11,27 @@ export class User {
...
+  lastName: string;
+
+  @Column({ type: 'varchar', default: 'default' })
+  phone: string;
+
+  @Column({ type: 'integer', default: 0 })
+  user: number;
+
+  @ManyToOne(() => ProfileEntity, (profile) => profile.user, { cascade: true })
+  userId: ProfileEntity;
```
Comment (lyudad) [inline src/entities/user.entity.ts]: again why we need user and iserId

File: src/main.ts [hunk 5]
```diff
@@ -1,9 +1,12 @@
 import { NestFactory } from '@nestjs/core';
 import { AppModule } from './app.module';
 import { ValidationPipe } from '@nestjs/common';
+import * as bodyParser from 'body-parser';
 
 async function bootstrap() {
   const app = await NestFactory.create(AppModule, { cors: true });
+  app.use(bodyParser.json({ limit: '50mb' }));
+  app.use(bodyParser.urlencoded({ limit: '50mb', extended: true }));
```
Comment (lyudad) [inline src/main.ts]: no magic numbers, please check general chat and my comment about magic numbers

File: src/entities/profile/profile.entity.ts (hunk 3 above)
Comment (Yulllia) [reply src/entities/profile/profile.entity.ts:57]: I need user Id to check equality between profile table and users

File: src/entities/user.entity.ts (hunk 4 above)
Comment (Yulllia) [reply src/entities/user.entity.ts]: I need user Id to check equality between profile table and users

File: src/main.ts (hunk 5 above)
Comment (Yulllia) [reply src/main.ts]: fixed it

File: src/modules/profile/profile-filter.dto.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  search?: string;
+
+  @ApiProperty({ required: false })
+  category?: string;
+
+  @ApiProperty({ required: false })
+  skills?: string;
+
+  @ApiProperty({ required: false })
+  page?: string;
```
Comment (lyudad) [inline src/modules/profile/profile-filter.dto.ts:14]: why we need page when create user?

File: src/modules/profile/profile-filter.dto.ts [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+  @ApiProperty({ required: false })
+  sort?: string;
```
Comment (lyudad) [inline src/modules/profile/profile-filter.dto.ts:17]: why we need sort when create user?

File: src/modules/profile/profile-filter.dto.ts (hunk 6 above)
Comment (Yulllia) [reply src/modules/profile/profile-filter.dto.ts:14]: I need page for pagination

File: src/modules/profile/profile-filter.dto.ts (hunk 7 above)
Comment (Yulllia) [reply src/modules/profile/profile-filter.dto.ts:17]: I use sort for sorting by price freelancer and its field not required only display query. Table display only query

Comment (Yulllia) [reply src/modules/profile/profile-filter.dto.ts:17]: I rename this table to findUserdto

File: src/modules/profile/profile-filter.dto.ts (hunk 6 above)
Comment (lyudad) [reply src/modules/profile/profile-filter.dto.ts:14]: now yes, but before DTO name was other

File: src/modules/profile/profile.controller.ts [hunk 8]
```diff
@@ -17,11 +18,63 @@ export class ProfileController {
...
+  @Get('allItem')
+  getAllProfile() {
+    return this.profileService.getAllProfile();
+  }
+
+  @Get('filter')
+  async findAll(@Query() userQuery: FindUserDto) {
+    const profileInfo = await this.profileService.queryBuilderSkills('skillsprofile');
+    const filterProfile = await this.profileService.paginationFilter(userQuery, profileInfo);
+    const user = await this.profileService.queryBuilderUser('userprofile');
```
Comment (lyudad) [inline src/modules/profile/profile.controller.ts]: replase with one query. its not optimal way of quering

Comment (Yulllia) [reply src/modules/profile/profile.controller.ts]: fixed

Comment (Yulllia) [reply src/modules/profile/profile.controller.ts]: fixed

**Pull request merged**